content/news/2021-07-01_haiku_inc_sponsoring_hardware_purchases_for_risc_v_port.md

```diff
@@ -0,0 +1,19 @@
++++
+type = "news"
+title = "Haiku, Inc sponsoring hardware purchases for RISC-V port"
+author = "leavengood"
+date = "2021-07-01 08:50:00+04:00"
+tags = ["RISC-V", "riscv64", "hardware", "haiku-inc", "donations"]
++++
+
+RISC-V is an interesting, relatively new and open source hardware platform. The inventors of RISC-V created a company called HiFive and that company recently released a desktop-class System on a Chip (SoC) board called [HiFive Unmatched](https://www.sifive.com/boards/hifive-unmatched). Long before this release Haiku developer Alexander von Gluck had pre-ordered this board and had begun work on the Haiku port to RISC-V, making some progress on the boot loader, u-boot support and memory mapping.
+
+Starting about two months ago Haiku developer Ilya Chugin, known in the community as X512, also started working on the RISC-V port for Haiku, but from another angle. All the details are too much for this post, but they can be read in [his associated Haiku forum post](https://discuss.haiku-os.org/t/my-haiku-risc-v-port-progress/10663). To summarize: he ported a small RISC-V emulator called [TinyEmu](https://bellard.org/tinyemu/) to Haiku, wrote a toy operating system and other tools to learn the platform, then he slowly got Haiku working in that emulator, with full GUI support, progressively getting more and more things working. He then started doing similar work in QEMU, which more accurately emulates real hardware. All of this was done in Haiku itself running on an x86 computer.
```

Comment:
Hi @X547 , are you ok with being referred by your real name in official news?

Reply:
I would avoid to be easily googled real name <-> nick name relation if possible. Real name is already mentioned in about dialog credits and some forum posts anyway. Proposed mentioning seems fine.

Reply:
I can make it all X512 here. Your real name is in the forum post asking for the sponsorship. I didn't even know your name myself before that 😄

Reply:
Maybe it will be better use X512 everywhere. I just want to keep internet life and real life separated.

Reply:
I will change it and amend the commit so there is less history here too.